pilz_trajectory_generation/test/unittest_trajectory_generator.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <memory>
+
+#include <gtest/gtest.h>
+
+#include <pilz_trajectory_generation/trajectory_generator.h>
+
+using namespace pilz;
+
+/**
+ * @brief Checks that each derived MoveItErrorCodeException contains the correct
+ * error code.
+ */
+TEST(TrajectoryGeneratorTest, TestExceptionErrorCodeMapping)
+{
+  std::shared_ptr<TrajectoryGeneratorInvalidLimitsException> tgil_ex {new TrajectoryGeneratorInvalidLimitsException("")};
```

Comment:
Given the variable naming to avoid issues you could introduce scope to avoid typos in future changes...

Reply:
Done.